thorlcr/slave/slavmain.cpp

```diff
@@ -1900,9 +1900,11 @@ class CJobListener : public CSimpleInterface
                         PROGLOG("Started wuid=%s, user=%s, graph=%s [log detail level=%u]", wuid.get(), user.str(), graphName.get(), maxLogDetail);
                         PROGLOG("Using query: %s", soPath.str());
 
-                        if (!getExpertOptBool("slaveDaliClient") && workUnitInfo->getPropBool("Debug/slavedaliclient", false))
+                        // slaveDaliClient option deprecated, but maintained for compatibility
+                        if ((!getExpertOptBool("allowDaliAccess") && workUnitInfo->getPropBool("Debug/allowdaliaccess", false)) ||
+                            (!getExpertOptBool("slaveDaliClient") && workUnitInfo->getPropBool("Debug/slavedaliclient", false)))
```

Comment:
I doubt it will cause a problem in practice, but this condition isn't quite right.  It should be

```
     if ((!getExpertOptBool("allowDaliAccess") && !getExpertOptBool("slaveDaliClient") && 
           (workUnitInfo->getPropBool("Debug/allowdaliaccess", false)) || (workUnitInfo->getPropBool("Debug/slavedaliclient", false)))
```

similar for the other tests.

Reply:
needs to be this actually I think (brackets enclosing the || terms):
```
                        if (!getExpertOptBool("allowDaliAccess") && !getExpertOptBool("slaveDaliClient") &&
                            (workUnitInfo->getPropBool("Debug/allowdaliaccess", false) || workUnitInfo->getPropBool("Debug/slavedaliclient", false)))
```